### PR TITLE
feat: roving focus package

### PR DESCRIPTION
**Pull request by alitnk**
#608 (PR: #642), #630 (PR: #631) and #607 (PR: #650) need this for keyboard navigation

### CODE REVIEW

**Comment by alitnk** [comment]
Sorry for the Vercel ping Ehsan. I noticed you've already ported `collection` so I just cherry picked your commit.

**Comment by alitnk** [comment]
@natew I think this is ready.

**Comment by natew** [comment]
I see `radix` in the code.

Not the most happy with the size of this, feels like collection is very abstraction-heavy. I think that and not sure what Slot is doing can't we just pass props? But I can merge this if you feel those are worth the cost.

**Comment by natew** [comment]
Yea just fix up conflicts and remove radix, I think it's ok. It doesn't seem to have any carve-out for native doesn't native not need this?

**Comment by alitnk** [comment]
> Not the most happy with the size of this, feels like collection is very abstraction-heavy. I think that and not sure what Slot is doing can't we just pass props? But I can merge this if you feel those are worth the cost.

I also agree it could probably be simpler. We can just change collection later as we don't use it directly (except for the accordion PR)

> It doesn't seem to have any carve-out for native doesn't native not need this?

The purpose of this package is focus management of groups by left-right or up-down buttons (you can see it in use in #631) - I don't know if there's an equivalent for that on native. But if you have an idea please let me know.

File: packages/roving-focus/types/RovingFocusGroup.d.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { Stack } from '@tamagui/core';
+import * as React from 'react';
+/**
+ * A custom hook that converts a callback to a ref to avoid triggering re-renders when passed as a
+ * prop or avoid re-executing effects when passed as a dependency
+ */
+declare function useCallbackRef<T extends (...args: any[]) => any>(callback: T | undefined): T;
```
Comment (amaranth) [inline packages/roving-focus/types/RovingFocusGroup.d.ts:7]: Isn't this basically the `useEvent` already defined in `@tamagui/use-event`?

Comment (alitnk) [reply packages/roving-focus/types/RovingFocusGroup.d.ts]: Replaced it with `useEvent`. Thanks for pointing out.

**Comment by natew** [comment]
Lets make `.native.ts` versions of these that exports empty functions.

**Comment by alitnk** [comment]
> Lets make `.native.ts` versions of these that exports empty functions.

@natew when porting components to tamagui, there are use cases like this:
``` 
<RovingFocusGroup asChild ...>
```
The current implementation allows these to work. I guess we could use a Slot instead of the Stack there.

We could also just return children, but it wouldn't be a 1-1 to web implementation anymore and we should probably only use it on `.web.ts` files at that point. It's your call though.

**Comment by natew** [comment]
I think it's fine as is then, I'll merge this unless you have any objection.

**Pull request merged**